plugin/src/functionalTest/kotlin/io/qalipsis/gradle/cloud/SSEListenerTests.kt

```diff
@@ -0,0 +1,364 @@
+package io.qalipsis.gradle.cloud
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.InjectMockKs
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.spyk
+import io.mockk.verify
+import io.qalipsis.gradle.cloud.listener.SSEListener
+import okhttp3.sse.EventSource
+import java.util.concurrent.CompletableFuture
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.gradle.api.logging.Logger
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.TestInstance.Lifecycle
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import org.junit.jupiter.params.provider.ValueSource
+import java.time.Instant
+import java.time.ZoneId
+import java.util.stream.Stream
+import kotlin.test.assertTrue
+
+@TestInstance(Lifecycle.PER_CLASS)
+internal class SSEListenerTests {
+
+    @MockK
+    lateinit var logger: Logger
+
+    private var operationResult = CompletableFuture<Result<Unit>>()
+
+    private val eventSource = mockk<EventSource>()
+
+    @InjectMockKs
+    private lateinit var sseListener: SSEListener
+
+    @BeforeAll
+    fun initMocks() {
+        MockKAnnotations.init(this)
+    }
+
+    @BeforeEach
+    fun initializeMocks() {
+        operationResult = spyk(CompletableFuture())
+        sseListener = spyk(SSEListener(operationResult, logger), recordPrivateCalls = true)
+    }
+
+    @Test
+    fun `should receive valid events and process them`() {
+        // given
+        val request = mockk<Request>(relaxed = true)
+        val data = "{ \"text\": \"Successfully published Scenario.. 0\"," +
```

Comment:
Please use raw literals for such strings. Kotlin provides facilities with trimIndent and trimMargin to make strings readable.